### PR TITLE
Space Lube TEG + Reaction Temperature Cost

### PR DESCRIPTION
**Pull request by Kurfursten**
This will make more sense if you read the changelog first.

You can only get toxic waste out by pouring in ethanol unless you barter for a baster/dropper, there's no special dump reagents feature. Don't let the clown fill it with water and vomit. Honk.

Added two new vars for chemical heating:
* reacts_discretely: If 1, the holder will never try to make a multiple of your recipe. For instance, if water/potass explosion had this var, it would always calculate explosion size as 1u potassium 1u water.
* reaction_temp_cost: Used if you want your reaction to burn off some heat in exchange for the conversion.

These two were combined so that I could cause only 1u of lube to convert to waste at a time, then it needs to heat up more to convert the next unit. You could use this system for other things, like making water slowly convert to ice instead of all at once.

Do you think such a thing as an ethanol tank should exist in engineering? It would basically function like an industrially useful beer keg. I'll add one if people like the idea.

🆑 
* rscadd: You may now lube up TEG circulators (up to 25u each) to increase power output. Power growth is linear up to 200% bonus based on flow capacity.
* rscadd: Lube in circulators will slowly heat until converting into toxic waste.
* rscadd: Degrease engine by pouring in ethanol. Ethanol cancels out toxic waste.

### CODE REVIEW

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 1]
```diff
@@ -18,6 +18,8 @@
 	var/list/secondary_results = list() //Additional reagents produced by the reaction
 	var/is_cold_recipe = 0
 	var/required_temp = 0
+	var/react_discretely = 0 //Handle_reactions() won't find the maximum number of times the chemicals can react. Use only if it shouldn't react more than once at a time.
```
Comment (DamianX) [inline code/modules/reagents/Chemistry-Recipes.dm]: replace with TRUE and FALSE

**Comment by DamianX** [comment]
Nice. You might want to update the wiki for the TEG.

**Comment by Kurfursten** [comment]
The TEG page as a whole is very overdue for a complete rewrite, which I'll probably do anyway after this hits.

**Comment by Kurfursten** [comment]
untitled

Ethanol tank is semitransparent. I was thinking just one on Box and Packed only, since they have TEGs. Space lube tank would be unspeakably terrifying, so no chance of a tank for _that_.

**Comment by PJB3005** [comment]
Deff has a TEG too by the R-UST.

**Comment by Fruktlimpa** [comment]
An actual proper use of space lube? Insanity.

**Review by PJB3005 — APPROVED** [review]

**Comment by Hugh-G-Rection** [comment]
Space lube tank. The three most beautiful words I've ever heard.
Name your price, bounty-hounds.

**Comment by Kurfursten** [comment]
I knew this Hugh post was coming as soon as I saw "Don't let the clown fill it with water 👍 " and "No space lube tank 👎 "

**Comment by Kurfursten** [comment]
Every time the map cleaner works, I consider it a personal miracle from Jesus Christ to me. Thank you Jesus.

💯 ready now

**Comment by Kurfursten** [comment]
I will update the wiki now, though a proper cleanup will take some time

**Pull request merged**